internal/btf/btf.go

```diff
@@ -176,16 +200,52 @@ func LoadKernelSpec() (*Spec, error) {
 }
 
 func loadKernelSpec() (*Spec, error) {
-	fh, err := os.Open("/sys/kernel/btf/vmlinux")
-	if os.IsNotExist(err) {
-		return nil, fmt.Errorf("can't open kernel BTF at /sys/kernel/btf/vmlinux: %w", ErrNotFound)
-	}
+	release, err := unix.KernelRelease()
 	if err != nil {
-		return nil, fmt.Errorf("can't read kernel BTF: %s", err)
+		return nil, err
+	}
+
+	locations := []struct {
+		pathFmt string
+		raw     bool
+	}{
+		/* try canonical vmlinux BTF through sysfs first */
+		{"/sys/kernel/btf/vmlinux", true /* raw BTF */},
+		/* fall back to trying to find vmlinux ELF on disk otherwise */
+		{"/boot/vmlinux-%s", false},
+		{"/lib/modules/%s/vmlinux-%s", false},
+		{"/lib/modules/%s/build/vmlinux", false},
+		{"/usr/lib/modules/%s/kernel/vmlinux", false},
+		{"/usr/lib/debug/boot/vmlinux-%s", false},
+		{"/usr/lib/debug/boot/vmlinux-%s.debug", false},
+		{"/usr/lib/debug/lib/modules/%s/vmlinux", false},
+	}
+
+	var spec *Spec
+	for _, loc := range locations {
+		path := fmt.Sprintf(loc.pathFmt, release)
+		if err := syscall.Access(path, linux.R_OK); err != nil {
```

Comment:
This seems strange, what's the point of doing this?

Reply:
I was mirroring what `libbpf` does, but it looks like `access` checks for access using the real UID/GID rather than effective. That doesn't seem relevant here. I'll remove it. https://www.man7.org/linux/man-pages/man2/access.2.html